### PR TITLE
Properly handle expression variable declared within VariableDeclaratorSyntax.ArgumentList in field declarations.

### LINKED ISSUE #13528: Adjust wording for ERR_ExpressionVariableInConstructorOrFieldInitializer error
```
+  <data name="ERR_ExpressionVariableInConstructorOrFieldInitializer" xml:space="preserve">
+    <value>Out variable or pattern variable declarations are not allowed within constructor/field/auto-implemented property initializers.</value>
```

From @cston:
Consider separating cases with comma for consistency with other errors. And remove auto-implemented if not necessary. Perhaps: ... not allowed within constructors, field initializers, or property initializers.

From @gafter:
Also, I believe the first "or" should be "and" (they are both forbidden), so I suggest

>Out variable and pattern variable declarations are not allowed within constructor initializers, field initializers, or property initializers.

This is long-winded enough that you could consider specializing it.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to #12597 and # 13417.
Also fixes #13528.

@dotnet/roslyn-compiler, @gafter Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
Ping @dotnet/roslyn-compiler, @gafter Please review.

**Comment by gafter** [comment]
:shipit:

**Comment by AlekseyTs** [comment]
Ping @dotnet/roslyn-compiler Please review, need a second sign-off.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs [hunk 1]
```diff
@@ -19578,5 +19578,460 @@ static bool TakeOutParam(object y, out bool x)
...
+        private static void VerifyModelNotSupported(
+            SemanticModel model,
+            DeclarationExpressionSyntax decl,
+            params IdentifierNameSyntax[] references)
+        {
+            var variableDeclaratorSyntax = GetVariableDesignation(decl);
+            Assert.Null(model.GetDeclaredSymbol(variableDeclaratorSyntax));
+            Assert.Null(model.GetDeclaredSymbol((SyntaxNode)variableDeclaratorSyntax));
+
+            Assert.False(model.LookupSymbols(decl.SpanStart, name: decl.Identifier().ValueText).Any());
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Consider extracting a local for `decl.Identifier().ValueText`.

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs [hunk 2]
```diff
@@ -14460,19 +14461,45 @@ unsafe struct S
...
                 );
         }
 
+        private static void VerifyModelNotSupported(SemanticModel model, DeclarationPatternSyntax decl, params IdentifierNameSyntax[] references)
+        {
+            Assert.Null(model.GetDeclaredSymbol(decl));
+            Assert.Null(model.GetDeclaredSymbol((SyntaxNode)decl));
+
+            Assert.False(model.LookupSymbols(decl.SpanStart, name: decl.Identifier.ValueText).Any());
+            Assert.False(model.LookupNames(decl.SpanStart).Contains(decl.Identifier.ValueText));
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs]: Consider extracting a local for `decl.Identifier.ValueText`.

**Comment by cston** [comment]
LGTM

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Sure, will include this change in the next PR.

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs (hunk 2 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs]: Sure, will include this change in the next PR.

**Pull request merged**